xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp

```diff
@@ -388,6 +388,7 @@ void CVideoPlayerAudio::Process()
       if (bPacketDrop)
       {
         m_syncState = IDVDStreamPlayer::SYNC_STARTING;
+        m_audioSink.Flush();
```

Comment:
This should work. I recommend to test accurate seeks for regression.

```
      if (bPacketDrop)
      {
        pMsg->Release();
        if (m_syncState != IDVDStreamPlayer::SYNC_STARTING)
        {
          m_audioSink.Drain();
          m_audioSink.Flush();
          audioframe.nb_frames = 0;
        }
        m_syncState = IDVDStreamPlayer::SYNC_STARTING;
        continue;
      }
```

Reply:
Thanks @FernetMenta I've adjusted based on your patch. This works similar to the original code - mutes are correctly applied. Tested accurate seeks and they work as expected.

Looking at the log and playing the sample I still get the `GENERAL_RESYNC` triggered a few ms after the EDL mute started. This happens because of the `PLAYER_STARTED` being triggered in VPA (snippet of my last comment). Furthermore while playing in the second mute zone I also still get the "large audio sync error" messages.

Is the result of `DEMUXER_PACKET` reception and the actual packet output supposed to be somehow in sync? I ask this because the packet that causes the `PLAYER_STARTED` message to be sent (the first packet with a dts > than the mute zone) comes right after the mute (the playback I mean) has actually started. Likely this is the way this is supposed to work and the log messages are harmless but still, you'd be better to confirm with you before considering merging this. Wouldn't this indicate some issue with the audio clock?

Snippet of the log here: https://pastebin.com/DBTxVvEM